config_src/nuopc_driver/time_utils.F90

```diff
@@ -166,4 +160,16 @@ function string_to_date(string, rc)
 
 end function string_to_date
 
+logical function chkerr(rc, line, file)
```

Comment:
This function is being duplicated in many modules. Is there a reason for this? If possible, I suggest keeping only one (public) copy of it so that it can be called from the other modules.